BraveWallet/Settings/BiometricsPasscodeEntryView.swift

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import BraveUI
+import struct Shared.Strings
+import SwiftUI
+
+struct BiometricsPasscodeEntryView: View {
+  
+  @ObservedObject var keyringStore: KeyringStore
+  
+  private enum UnlockError: LocalizedError {
+    case incorrectPassword
+    
+    var errorDescription: String? {
+      switch self {
+      case .incorrectPassword:
+        return Strings.Wallet.incorrectPasswordErrorMessage
+      }
+    }
+  }
+  
+  @State private var password: String = ""
+  /// Error occured when user entered their password
+  @State private var unlockError: UnlockError?
+  /// Flag used to determine if we should show an error alert because we failed to store the password in the keychain
+  @State private var isShowingKeychainError: Bool = false
+  @Environment(\.presentationMode) @Binding private var presentationMode
+  
+  private var isPasswordValid: Bool {
+    !password.isEmpty
+  }
+  
+  private func unlock() {
```

Comment:
Is there a way we can call unlock method with a password parameter so we don't have to store password in a property?